src/components/Profile.tsx

```diff
@@ -0,0 +1,292 @@
+import {
+  Box,
+  Button,
+  Container,
+  CssBaseline,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  InputAdornment,
+  TextField,
+  ThemeProvider,
+  Typography,
+  createTheme,
+} from "@mui/material";
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { setCurrentUser } from "../app/slices/UserSlice";
+import { sdk } from "../config/sharetribeSDK.config";
+import ArrowForwardIcon from "@mui/icons-material/ArrowForward";
+import { Edit } from "@mui/icons-material";
+import { ChangeImage } from "./AddOrChangeImage";
+
+const defaultTheme = createTheme();
+
+export const Profil = (props: any) => {
+  const currentUser = useSelector((state: any) => state.user.currentUser);
+  const [file, setfile] = useState<any>();
+  const [user, setUser] = useState({
+    firstName: currentUser.data?.attributes.profile.firstName,
+    lastName: currentUser.data?.attributes.profile.lastName,
+    displayName: currentUser.data?.attributes.profile.displayName,
+    privateData: currentUser.data?.attributes.profile.privateData,
+    protectedData: currentUser.data?.attributes.profile.protectedData,
+    publicData: currentUser.data?.attributes.profile.publicData,
+    bio: currentUser.data?.attributes.profile.bio,
```

Comment:
i think you not need open one one, you can:
const const [user, setUser] = useState({
...currentUser.data?.attributes.profile
})

try it.

Reply:
ניסית לעשות את מה שכתבתי?

Reply:
זה לא יעזור,כי ה currenyUser מכיל עוד שדות בפרופיל שלא נכללים בפרופיל שמעדכנים(כגון מייל, abbreviatedName)